refactor(product_modify): product, productLike 구조 변경 및 review, best 엔티티 정의

Ko-udon opened the pull request:
## 변경사항

<!-- 이 PR에서 어떤점들이 변경되었는지 기술해주세요.  -->

1. product -> productLike  일대일 단방향 구조로 변경

2. 별도의 best 테이블 생성

3. review 엔티티 생성


## 체크 리스트 (Checklist)

pull request 전에 아래 체크 리스트들을 만족하는 지 확인한 후 체크('x') 표시를 해주시기 바랍니다.

- [x]  master 브랜치가 아니라 **develop** 브랜치에 Merge하도록 pull request를 작성 중이신가요?
- [x]  모든 **테스트**가 성공했나요?

ehddn committed:
refactor(product_modify): product, productLike 구조 변경 및 review, best 엔티티 생성

1. product -> productLike  일대일 단방향 구조로 변경

2. 상품의 카테고리별로 맵핑되는 INT값 기획안 반영

3. 별도의 best 테이블 생성

4. review 엔티티 생성

seyoung5744 commented on src/main/java/com/zerobase/wishmarket/domain/product/model/entity/Best.java:
```diff
@@ -0,0 +1,26 @@
+@AllArgsConstructor
+@NoArgsConstructor
+@Entity
+public class Best {
```
best 상품의 경우 하루 단위로 업데이트가 이뤄지기 때문에 RDB에 table로 관리하는건 비효율적인거 같습니다.
하루 단위로 사용되는 데이터니깐 Redis를 이용하시는 건 어떨까요

Kim-SeonBeom replied:
구현을 우선 목표로 하고 추후 리팩토링할때 시도해보려합니다

seyoung5744 commented on src/main/java/com/zerobase/wishmarket/domain/product/model/entity/ProductLikes.java:
```diff
@@ -3,30 +3,26 @@
 @Setter
+@Getter
 @AllArgsConstructor
-@NoArgsConstructor(access = AccessLevel.PROTECTED)
```
기본 생성자 접근을 최대한 막을려고 설정했던 속성인데 어떤 이유로 지우셨나요?

seyoung5744 commented on src/main/java/com/zerobase/wishmarket/domain/product/model/entity/Review.java:
```diff
@@ -0,0 +1,33 @@
+
+    private boolean isRecommend;
+
+    private Long ProductId;
```
변수명 시작은 소문자로 해주세요

seyoung5744 commented on src/main/java/com/zerobase/wishmarket/domain/product/model/entity/Review.java:
```diff
@@ -0,0 +1,33 @@
+import lombok.Setter;
+
+@Getter
+@Setter
```
@Setter는 최후의 수단으로 생각해주세요

Kim-SeonBeom replied:
테스트코드때문에 달아놓은걸 안지웠네요 수정하겠습니다

seyoung5744 commented on src/main/java/com/zerobase/wishmarket/domain/product/model/entity/Review.java:
```diff
@@ -0,0 +1,33 @@
+@AllArgsConstructor
+@Builder
+@Entity
+public class Review {
```
BaseEntity를 상속받아서 리뷰 생성일 관리도 같이 해주세요

Kim-SeonBeom replied:
ERD에 없어서 추가하지않았었습니다 수정하겠습니다

ehddn committed:
fix(product_modify): product, productLikes, review 수정

1. productLikes에 @NoArgsConstructor(access = AccessLevel.PROTECTED) 접근제한설정

2. review
- 변수명 소문자로 변경
- @Setter 제거
- BaseEntity 상속
